.github/scripts/determine-runners.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+require "formula"
+
+formula = Formula[ENV["HOMEBREW_FORMULA_TO_REBOTTLE"]]
+timeout = ENV["HOMEBREW_REBOTTLE_TIMEOUT"].to_i
+
+linux_runner = if timeout > 360
+  "linux-self-hosted-1"
+else
+  "ubuntu-latest"
+end
+linux_runner_spec = {
+  runner:    linux_runner,
+  container: {
+    image:   "ghcr.io/homebrew/ubuntu16.04:master",
+    options: "--user=linuxbrew",
+  },
+  workdir:   "/github/home",
+}
+
+tags = formula.bottle_specification.collector.tags
+runners = if tags.count == 1 && tags.first.system == :all
+  # Build on all supported macOS versions and Linux.
+  MacOS::Version::SYMBOLS.values.flat_map do |version|
+    macos_version = MacOS::Version.new(version)
+    if macos_version.outdated_release? || macos_version.prerelease?
+      nil
+    else
+      macos_runners = [{ runner: macos_version.to_s }]
+      macos_runners << { runner: "#{macos_version}-arm64" } if macos_version >= :big_sur
+      macos_runners
+    end
+  end << linux_runner_spec
+else
+  tags.map do |tag|
+    macos_version = tag.to_macos_version
+
+    if macos_version.outdated_release?
+      nil # Don't rebottle for older macOS versions (no CI to build them).
+    else
+      runner = macos_version.to_s
+      runner += "-#{tag.arch}" unless tag.arch == :x86_64
+      { runner: runner }
+    end
+  rescue MacOSVersionError
+    if tag.system == :linux && tag.arch == :x86_64
+      linux_runner_spec
+    elsif tag.system == :all
+      # An all block with OS-specific bottles also present.
+      # This shouldn't happen unless there's a bug elsewhere.
+      nil
```

Comment:
We have a few of these; they're not always bugs. Some are formulae first bottled before we had `:all` bottles. Though we should probably be rebottling those.

Reply:
I think I fixed it so an `:all` bottle cannot be added if there are already OS-specific bottles.

The opposite maybe could happen, but really shouldn't.

Reply:
Yes, I was referring to formulae that were bottled in the period after multiple-runner `dispatch-build-bottle` but before you made that change.

Reply:
Those were consequences of bugs so I'd still say it's a bug they're even still present but I'll remove the comment anyway.

Reply:
It's still a slightly useful bug, though, as it identified some formulae that actually should have `:all` bottles. But it was wrong in some cases.

Reply:
You may find this useful:

```rb
Formula.each do |f|
  monterey_spec = f.bottle_specification
                   .tag_specification_for(Utils::Bottles.tag(:monterey), no_older_versions: true)
  arm64_monterey_spec = f.bottle_specification
                         .tag_specification_for(Utils::Bottles.tag(:arm64_monterey), no_older_versions: true)
  puts f.name if monterey_spec && arm64_monterey_spec &&
                 monterey_spec.tag != arm64_monterey_spec.tag &&
                 monterey_spec.checksum == arm64_monterey_spec.checksum
end
```

I'd always test `:all` rebottles in a PR first however, to make sure they actually do produce `:all` bottles. Also watch for `uses_from_macos ..., since: :catalina` since we're not checking Mojave and older anymore.

Reply:
Oh, nice. I'm more interested in cases where the Linux bottle has the same checksum as an ARM bottle though. Usually those are just cases where keg relocation is replacing `/usr/local` references it doesn't need to.

Reply:
Replace `:monterey` with `:x86_64_linux` above and it should work, though the bottles being built at different times may impact it due to reproducibility improvements we've since made (e.g. `TZ`).

Reply:
Yep, exactly what I did. There are ~60 such formulae. There are ~30 formulae with the same ARM/X86 macOS checksums. (Probably some overlap.)